pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -120,6 +120,9 @@ func (arrayOrString *ArrayOrString) UnmarshalJSON(value []byte) error {
 	switch value[0] {
 	case '[':
 		arrayOrString.Type = ParamTypeArray
+		// We're tring to Unmarshal to []string, but for cases like []int or other types
+		// of nested array which we don't support yet, we should continue and Unmarshal
+		// it to String.
```

Comment:
Can you also mention something like this "if the Type being set doesn't match what it actually should be, it will be captured somewhere else"